gateway/downlink.go

```diff
@@ -123,7 +125,8 @@ func accurateSleep(ctx context.Context, duration time.Duration) bool {
 // Downlink payload structure
 // | MHDR | DEV ADDR | FCTRL | FCNTDOWN |  FOPTS (optional)  |  FPORT | encrypted frame payload  |  MIC |
 // | 1 B  |   4 B    |  1 B  |    2 B   |       variable     |   1 B  |      variable            | 4 B  |
-func (g *gateway) createDownlink(device *node.Node, framePayload []byte, sendAck bool, uplinkFopts []byte) ([]byte, error) {
+func (g *gateway) createDownlink(device *node.Node, framePayload []byte, sendAck bool, uplinkFopts []byte, snr float64, sf int) (
+	[]byte, error) {
```

Comment:
Remember you can group variables of similar types together like so for readability. Okay to leave as-is for now.
```suggestion
func (g *gateway) createDownlink(device *node.Node, framePayload, uplinkFopts []byte, sendAck bool, snr float64, sf int) (
	[]byte, error) {
```